### PR TITLE
Filter additon in the post and patch api request and unittestcases correction

### PR DESCRIPTION
**Pull request by denduluriramasaimohanakhil**
dev:  
lint:0
  coverage:85%

AdditionalPostPatchFilter:
  lint:0
  coverage:85%

### CODE REVIEW

File: src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java [hunk 1]
```diff
@@ -161,7 +159,7 @@ void saveFormDataNewCollectionTest() throws IOException
         Mockito.when(mockFormValidationServiceImpl.validateData(any(),any(),anyString())).thenReturn(validationResultList);
         Mockito.when(mockObjectMapper.convertValue(any(),eq(FormDataDefinition.class))).thenReturn(new FormDataDefinition());
         Mockito.when(mockUserDetails.getUserDetails()).thenReturn(userList);
-        mockFormDataServiceImpl.saveFormData(formDataSchema);
+        mockFormDataServiceImpl.saveFormData(formDataSchema,"formData.name:akhil");
```
Comment (VaibhavTechsophy) [inline src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java:161]: please write proper test cases to verify that a filter is translated and sent to downstream.

File: src/test/java/com/techsophy/tsf/runtime/form/service/impl/FormDataServiceImplTest.java [hunk 2]
```diff
@@ -49,7 +49,7 @@ void updateFormDataTestWhileIdIsEmpty()
         LinkedHashMap<String,Object> map = new LinkedHashMap<>();
         map.put(STRING,STRING);
         FormDataSchema formDataSchema=new FormDataSchema(TEST_ID,TEST_FORM_ID,TEST_VERSION,TEST_FORM_DATA,TEST_FORM_META_DATA);
-        Assertions.assertThrows(InvalidInputException.class, () -> formDataService.updateFormData(formDataSchema));
+        Assertions.assertThrows(InvalidInputException.class, () -> formDataService.updateFormData(formDataSchema,"formData.name:akhil"));
```
Comment (VaibhavTechsophy) [inline src/test/java/com/techsophy/tsf/runtime/form/service/impl/FormDataServiceImplTest.java:52]: please write proper test case to verify that an appropriate filter is added to downstream

File: src/main/java/com/techsophy/tsf/runtime/form/service/impl/FormDataServiceImpl.java [hunk 3]
```diff
@@ -120,8 +121,11 @@ public FormDataDefinition saveFormData(FormDataSchema formDataSchema) throws IOE
                 }
                 else
                 {
-                    Query query=new Query(Criteria.where(UNDERSCORE_ID).is(formDataSchema.getId()));
+                    Query query=StringUtils.isNotBlank(filter)?new Query(new Criteria().andOperator(Criteria.where(UNDERSCORE_ID).is(formDataSchema.getId()),
+                            getCriteria(filter))):new Query(Criteria.where(UNDERSCORE_ID).is(formDataSchema.getId()));
                     Optional<FormDataDefinition> existingFormDataDefinition=Optional.ofNullable(mongoTemplate.findOne(query,FormDataDefinition.class,TP_RUNTIME_FORM_DATA + formId));
+                    Objects.requireNonNull(formDataDefinition).setCreatedById(existingFormDataDefinition.orElseThrow().getCreatedById());
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/runtime/form/service/impl/FormDataServiceImpl.java]: what happens if the form data definition is not found?

**Review by VaibhavTechsophy — CHANGES_REQUESTED** [review]

File: src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java (hunk 1 above)
Comment (denduluriramasaimohanakhil) [reply src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java:161]: Yes writing testcases to verify filter

File: src/test/java/com/techsophy/tsf/runtime/form/service/impl/FormDataServiceImplTest.java (hunk 2 above)
Comment (denduluriramasaimohanakhil) [reply src/test/java/com/techsophy/tsf/runtime/form/service/impl/FormDataServiceImplTest.java:52]: Yes writing proper test case for appropriate filter

File: src/main/java/com/techsophy/tsf/runtime/form/service/impl/FormDataServiceImpl.java (hunk 3 above)
Comment (denduluriramasaimohanakhil) [reply src/main/java/com/techsophy/tsf/runtime/form/service/impl/FormDataServiceImpl.java]: checking for not null conditions and throwing 400 bad request for invalid input

**Review by VaibhavTechsophy — APPROVED** [review]

**Comment by VaibhavTechsophy** [comment]
pls deploy

**Pull request merged**